src/megaapi_impl.cpp

```diff
@@ -2773,9 +2778,12 @@ void MegaTransferPrivate::setLastBytes(char *lastBytes)
     this->lastBytes = lastBytes;
 }
 
-void MegaTransferPrivate::setLastError(MegaError e)
+void MegaTransferPrivate::setLastError(const MegaError *e)
 {
-    this->lastError = e;
+    if (lastError)
+        delete lastError;
+
+    lastError = e ? static_cast<MegaErrorPrivate*>(e->copy()) : nullptr;
```

Comment:
static_cast is ok for going up an inheritance heirarchy tree, but for going deeper into the tree we should use dynamic_cast, in case the object passed in is not of that type - eg. just a plain MegaError, not the private version.

Reply:
Agree

Reply:
Solved